src/components/Table/AutoDataTable/utils.js

```diff
@@ -0,0 +1,323 @@
+        }
+        break
+    }
+    // this.$log.debug('Field: ', type, col.prop, col)
+    return col
+  }
+
+  setDefaultFormatterIfNeed(col) {
+    const h = this.vm.$createElement
+    if (!col.formatter) {
+      col.formatter = (row, column, cellValue) => {
+        let value = cellValue
+        let padding = '0'
+        const excludes = [undefined, null, '']
+        if (excludes.indexOf(value) !== -1) {
+          padding = '6px'
+          value = '-'
+        }
+        return h('span', {
+          'style': {
+            marginLeft: padding
+          }
+        }, [value])
+      }
+    }
+    return col
+  }
+
+  setDefaultWidthIfNeed(col) {
+    const lang = i18n.locale
+    let factor = 10
+    if (lang === 'zh') {
+      factor = 20
+    }
+    let [sortable, filters] = [0, 0]
+    if (col && col?.sortable === 'custom') {
+      sortable = 10
+    }
+    if (col && col?.filters?.length > 0) {
+      filters = 12
+    }
+    if (col && !col.width && col.label && !col.minWidth) {
+      col.minWidth = `${col.label.length * factor + sortable + filters + 30}px`
+    }
+    return col
+  }
+
+  addOrderingIfNeed(col) {
+    if (col.prop) {
+      const column = this.meta[col.prop] || {}
+      if (column.order) {
+        col.sortable = 'custom'
+        col['column-key'] = col.prop
+      }
+    }
+    return col
+  }
+
+  addHelpTipIfNeed(col) {
+    const helpTip = col.helpTip
+    if (!helpTip) {
+      return col
+    }
+    col.renderHeader = (h, { column, $index }) => {
+      const binds = {
+        props: {
+          placement: 'bottom',
+          effect: 'dark',
+          openDelay: 500,
+          popperClass: 'help-tips'
+        }
+      }
+
+      return (
+        <span>{column.label}
+          <el-tooltip {...binds}>
+            <div slot='content' v-sanitize={helpTip}/>
+            <i class='fa fa-question-circle-o help-tip-icon' style='padding-left: 2px'/>
+          </el-tooltip>
+        </span>
+      )
+    }
+    return col
+  }
+
+  addFilterIfNeed(col) {
+    if (col.prop) {
+      const column = this.meta[col.prop] || {}
+      if (!column.filter) {
+        return col
+      }
+      if (column.type === 'boolean') {
+        col.filters = [
+          { text: i18n.t('Yes'), value: true },
+          { text: i18n.t('No'), value: false }
+        ]
+        col.sortable = false
+        col['column-key'] = col.prop
+      }
+      if (column.type === 'choice' && column.choices) {
+        col.filters = column.choices.map(item => {
+          if (typeof (item.value) === 'boolean') {
+            if (item.value) {
+              return { text: item['label'], value: 'True' }
+            } else {
+              return { text: item['label'], value: 'False' }
+            }
+          }
+          return { text: item['label'], value: item.value }
+        })
+        col.sortable = false
+        col['column-key'] = col.prop
+      }
+    }
+    return col
+  }
+}
```

Comment:
The comments were written using Chinese. The code does not appear to contain any significant errors other than minor typo corrections and missing parentheses or square brackets. The functions seem to be correctly documented.

One optimization suggestion can be:

```javascript
const hasColumnActions = configColumns.findIndex(item => item?.prop === 'actions') !== -1

// ... Rest of the script...
```

In this line, we find an action column by its prop but it might be beneficial if these checks could also match the action's visibility instead; you can modify the `actions` key in the `.config.columns` array to include the actions field that should never display when editing a record directly through the table.
For example:
```less
  if (!row.actionVisible(actions)) {
```